Control coverage/Feature/AVDApplicationGroups.md

```diff
@@ -0,0 +1,99 @@
+﻿# AVDApplicationGroups
+
+**Resource Type:** Microsoft.DesktopVirtualization/ApplicationGroups
+
+<!-- TOC -->
+
+- [Azure_AVDApplicationGroups_Audit_Enable_Diagnostic_Settings](#azure_AVDApplicationGroups_audit_enable_diagnostic_settings)
+
+<!-- /TOC -->
+<br/>
+
+___ 
+
+## Azure_AVDApplicationGroups_Audit_Enable_Diagnostic_Settings 
+
+### Display Name 
+Diagnostic settings must be enabled for Azure AVD Application Groups.
+
+### Rationale 
+Diagnostic logs are needed for creating activity trail while investigating an incident or a compromise.
+
+### Control Settings 
+```json 
+{
+    "DiagnosticForeverRetentionValue": "0",
+    "DiagnosticLogs": [
+        "Checkpoint",
+ 		"Management",
+ 		"Error"
+    ],
+    "DiagnosticMinRetentionPeriod":
+	"365"
+}
+ ``` 
+### Control Spec 
+
+> **Passed:** 
+> 1. Required diagnostic logs are enabled.
+> 2. At least one of the below setting configured:
+> a. Log Analytics.
+> b. Storage account (with min Retention period of 365 or forever(Retention period 0).
+> c. Event Hub.
+> 
+> **Failed:** 
+> 1. Diagnostics setting is disabled for resource.
+> 
+>       or
+>
+> 2. Diagnostic settings meet the following conditions:
+> a. All diagnostic logs are not enabled.
+> b. None of the below setting is configured:
+> i. Log Analytics.
+> ii. Storage account (with min Retention period of 365 or forever(Retention period 0).
+> iii. Event Hub.
+> 
+> **Error:** 
+> Required logs are not configured in control settings.
+> 
+
+### Recommendation 
+
+- **Azure Portal** 
+
+	 You can create or update the diagnostic settings from the Azure Portal by following the steps given here: https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings?tabs=portal#create-diagnostic-settings.
+
+- **PowerShell** 
+
+	 ```powershell 
+
+	# Below commands will be useful to Configure diagnostic settings on AVDApplicationGroups
+
+    Connect-AzAccount
+	Set-AzContext -SubscriptionId "<sub id>"`
+	
+	#Add diagnostic settings to an existing AVDApplicationGroups
+	$log = @()
+	$log += New-AzDiagnosticSettingLogSettingsObject -Enabled $true -CategoryGroup allLogs
+
+	New-AzDiagnosticSetting -Name <DiagnosticSettingName> -ResourceId <AVDApplicationGroupsResourceId> -WorkspaceId <LogAnalyticsWorkspaceID> -Log $log -StorageAccountId <StorageAccountResourceId> -EventHubName <EventHubName>-EventHubAuthorizationRuleId <EventHubAuthorizationRuleId>
```

Comment:
Shall we also add option for enabling Retention Days?

Reply:
Configuring retention days from diagnostic settings is not supported now. I have tried with Azure portal as well as PS commands but it clearly says not supported. please refer below screenshots as reference:

![image](https://github.com/azsk/AzTS-docs/assets/109667731/750ef427-0d69-45b7-93b1-ff96fa5972a3)

![image](https://github.com/azsk/AzTS-docs/assets/109667731/c597db3b-72e5-47d2-a7e3-3fcb4f08bdea)